### PR TITLE
Multiple media vue component

### PR DESCRIPTION
**Pull request by lawniczek-piotr**
I created multiple media component, which is compatible with asgard media linker (data sent to server has same structure as in old blade @media directives). However there is no ordering at this moment. Here is how to include it in your view:
1. `import MultipleFileSelector from '../../../../Media/Assets/js/mixins/MultipleFileSelector';`

2. Add `MultipleFileSelector` into your `mixins` array

3. Add component in your view:
eg.
```
<multiple-media zone="gallery" @fileUnselected="unselectFile($event, 'page')" @multipleFileSelected="selectMultipleFile($event, 'page')"
entity="Modules\Page\Entities\Page" :entity-id="$route.params.pageId"></multiple-media>   
```

### CODE REVIEW

**Comment by JulienTant** [comment]
Julien was here

**Comment by nWidart** [comment]
Hello & thank you, I'll have to test this out.

One comment for now:

- Please ignore your .idea folder in your global gitignore file

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,6 @@
 Homestead.yaml
 Homestead.json
 .env
+.php_cs.cache
 composer.lock
 package-lock.json
-.php_cs.cache
```
Comment (nWidart) [inline .gitignore]: Why was this moved?

**Comment by nWidart** [comment]
- Why did you compile the adminlte theme?

**Comment by lawniczek-piotr** [comment]
I didn't change .gitignore so i know there are a lot of things which shouldn't be changed in this pull request however i needed it fast in my current project so it may be a little bit messy. I probably compiled the adminlte theme by the habit as i use other template which has to be compiled while project is being installed.

**Comment by nWidart** [comment]
Hm ok, the diff says a line was moved in the gitignore file.

Don't forget to remove all files with `.idea` too.

**Comment by lawniczek-piotr** [comment]
I'll clear this later this evening or maybe tomorrow and make another pull request, I'm tight with time now.

**Comment by lawniczek-piotr** [comment]
Cleared changes a little bit. I've met some problems when component data is fetched before our entity model on view (eg. page) so it's good to use some v-if to make uselectFile work properly.

This is how to include component:

```
<multiple-media zone="gallery" @fileUnselected="unselectFile($event, 'page')" @multipleFileSelected="selectMultipleFile($event, 'page')"
entity="Modules\Page\Entities\Page" :entity-id="$route.params.pageId"></multiple-media>   
```

Offtopic: haven't you consider migarting to Vuex as you use Vue.js already? It would make data management easier

File: .gitignore [hunk 2]
```diff
@@ -7,3 +7,6 @@ Homestead.json
 composer.lock
 package-lock.json
 .php_cs.cache
+/public
```
Comment (nWidart) [inline .gitignore]: Do not ignore public please.

File: .gitignore [hunk 3]
```diff
@@ -7,3 +7,6 @@ Homestead.json
...
+/Themes
```
Comment (nWidart) [inline .gitignore]: Same here.

**Comment by nWidart** [comment]
Hello,

Sorry for the late reply!
I finally had time to test this.
I followed you steps, however, when adding 2 images on my page, they were not saved, when going back to edit.

**Comment by nWidart** [comment]
Oh ok, my bad I used your example from the first message to include it, but you changed it in your latest reply. 👍 

New issue:

- I added 2 images on a page
- I went back to the page, removed one image, clicked save.
- The 2 images are still linked.

**Comment by nWidart** [comment]
This is the console output:

```
2
app.js:1312 [Vue warn]: Error in event handler for "fileUnselected": "TypeError: Cannot read property 'gallery' of undefined"

found in

---> <MultipleMedia> at Modules/Media/Assets/js/components/MultipleMedia.vue
       <ElForm>
         <PageForm> at Modules/Page/Assets/js/components/PageForm.vue
           <Root>
warn @ app.js:1312
logError @ app.js:2445
globalHandleError @ app.js:2440
handleError @ app.js:2429
Vue.$emit @ app.js:3249
unSelectMedia @ app.js:117205
boundFn @ app.js:924
click @ app.js:117295
invoker @ app.js:2736
fn._withTask.fn._withTask @ app.js:2534
app.js:2449 TypeError: Cannot read property 'gallery' of undefined
    at VueComponent.unselectFile (app.js:117417)
    at Proxy.boundFn (app.js:923)
    at fileUnselected (app.js:108782)
    at VueComponent.invoker (app.js:2736)
    at VueComponent.Vue.$emit (app.js:3247)
    at VueComponent.unSelectMedia (app.js:117205)
    at Proxy.boundFn (app.js:924)
    at click (app.js:117295)
    at invoker (app.js:2736)
    at HTMLSpanElement.fn._withTask.fn._withTask (app.js:2534)
logError @ app.js:2449
globalHandleError @ app.js:2440
handleError @ app.js:2429
Vue.$emit @ app.js:3249
unSelectMedia @ app.js:117205
boundFn @ app.js:924
click @ app.js:117295
invoker @ app.js:2736
fn._withTask.fn._withTask @ app.js:2534
```

**Comment by nWidart** [comment]
I'll close this as there hasn't been a reply in a while, feel free to comment if needed.
